apps/frontend/src/app/5_pages/ProtocolDataPage/components/EcosystemStatistics/components/AmmContracts/AmmContracts.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { Decimal } from '@sovryn/utils';
+
+import { AmountRenderer } from '../../../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { decimalic } from '../../../../../../../utils/math';
+import { AmmLiquidityPoolDictionary } from '../../../../../MarketMakingPage/utils/AmmLiquidityPoolDictionary';
+import {
+  getConvertedValue,
+  getCurrencyPrecision,
+} from '../../../../../PortfolioPage/components/ProtocolSection/ProtocolSection.utils';
+import { EcosystemContracts } from '../../EcosystemStatistics.types';
+import {
+  PoolData,
+  EcosystemStatisticsProps,
+} from '../../EcosystemStatistics.types';
+import { AmmContractLiquidity } from './components/AmmContractLiquidity/AmmContractLiquidity';
+
+const ammPools = AmmLiquidityPoolDictionary.list();
+
+export const AmmContracts: FC<EcosystemStatisticsProps> = ({
+  selectedCurrency,
+  btcPrice,
+  onChange,
+}) => {
+  const [poolsData, setPoolsData] = useState<PoolData>({});
+
+  const onValueChange = useCallback((asset: string, balance: Decimal) => {
+    setPoolsData(prevBalances => {
+      const newBalances = { ...prevBalances, [asset]: balance };
+      return newBalances;
+    });
+  }, []);
+
+  const calculateBalance = useCallback(
+    () =>
+      Object.values(poolsData).reduce(
+        (total, balance) => total.add(balance),
+        Decimal.ZERO,
+      ),
+    [poolsData],
+  );
```

Comment:
Should be a useMemo beside that this can be removed and moved into renderBalance.